src/Tokenizer.js

```diff
@@ -457,52 +457,52 @@ module.exports = class Tokenizer {
   link(src) {
     const cap = this.rules.inline.link.exec(src);
     if (cap) {
-      const trimmedUrl = cap[2].trim();
-      if (!this.options.pedantic && trimmedUrl.startsWith('<')) {
-        // commonmark requires matching angle brackets
-        if (!trimmedUrl.endsWith('>')) {
-          return;
-        }
-
-        // ending angle bracket cannot be escaped
-        const rtrimSlash = rtrim(trimmedUrl.slice(0, -1), '\\');
-        if ((trimmedUrl.length - rtrimSlash.length) % 2 === 0) {
-          return;
-        }
-      } else {
+      let href = cap[2].trim();
+      if (!href.startsWith('<')) {
         // find closing parenthesis
         const lastParenIndex = findClosingBracket(cap[2], '()');
         if (lastParenIndex > -1) {
           const start = cap[0].indexOf('!') === 0 ? 5 : 4;
           const linkLen = start + cap[1].length + lastParenIndex;
-          cap[2] = cap[2].substring(0, lastParenIndex);
+          href = cap[2].substring(0, lastParenIndex);
           cap[0] = cap[0].substring(0, linkLen).trim();
           cap[3] = '';
         }
       }
-      let href = cap[2];
-      let title = '';
+
+      let title = cap[3] ? cap[3].slice(1, -1) : '';
+
       if (this.options.pedantic) {
+        // pedantic allows starting angle bracket without ending angle bracket
+        if (href.startsWith('<') && !href.endsWith('>')) {
```

Comment:
What if it has brackets at the beginning and end?
We don't want to remove just the starting bracket if there is a title.

Reply:
This does not seem to be easier to read. The logic seems out of order.

Reply:
I can understand if it just confuses things, especially since this probably screws with the mindset you had figuring out all of this out. If this isn't helping it doesn't need to be committed.

If it helps, essentially what I was trying to do was change the logic flow from this:
```
if(!A && B) { if (!C) {} }
if(A) {}
if(B { if(A && !C) {} }
```

to this:

```
if(!B) {}
if(A && B) {}
if(!A && B) {}
```

In other words:
1) In the absence of `<`, Process any `( )`
otherwise
2) If pedantic, handle `<>` in it's special way, including other pedantic-specific stuff
3) If not pedantic, handle `<>` in the Commonmark way

Reply:
I don't think that is any easier to read and testing `if(... B)` multiple times will be slower even if it is just slightly.

Reply:
I would rather read:

```
if (B) {
  if (A){
   ...
  } else {
   ...
  }
}
```

than

```
if (A && B){}
if (!A && B){}
```

Reply:
Ok. I think I still want to tinker with this, but otherwise if you want to just merge your PR as it is I can approve it. I might re-submit this as a separate PR later.

Reply:
The way I read the tokenizers is parsing the string from left to right.

```
deal with start of cap[1]
deal with endof cap[1]
deal with start of cap[2]
deal with endof cap[2]
etc.
```

Reply:
I'm ok with PRs that make it more readable as long as you keep `padantic` in mind.  It would be nice to be able to have a test suite just for pedantic.

Reply:
Also readability should not trump speed. Use comments if the code can't be more readable without making it slower.

Reply:
The main motivation for this was actually to reduce the number of checks for the Pedantic option in hopes of a speed boost by eliminating redundant checks, with a side effect of maybe being easier to read (clearly that did not happen). It looked to me like the code was running at about the same speed as before, because it ended up just trading off redundancy in one area for redundancy in another. Future changes to the List tokenizer will probably evolve how this is structured anyway, so we can take another look at it then.

As for Pedantic, this was still passing all the tests, so I assumed Pedantic was still working. Is there a good list of Pedantic test cases out there? I would be happy to start compiling up a spec file similar to the commonmark and gfm specs that already exist. Or do the `Original` tests already cover everything?

Reply:
> Is there a good list of Pedantic test cases out there?

We don't have very many pedantic specs and I haven't seen anything around. I think we will just have to make our own based on the [basics](https://daringfireball.net/projects/markdown/basics) and [syntax](https://daringfireball.net/projects/markdown/syntax) docs

Reply:
At one point we talked about moving pedantic to an extension instead of having it in core since there aren't too many people that use it.

Reply:
Are there any good extensions out there? That I could reference as a model for building new ones and get myself up to speed? I would be in favor of an extension for Pedantic.

Reply:
I don't think there are any plugins but it should just be an object that can be sent to `marked.use(extension)`

Something like:

```js
modules.exports = {
  tokenizer: {
    ...
  },
  renderer: {
    ...
  },
}
```

Reply:
Actually I forgot we do have the `/test/original` tests that are all run with `pedantic: true` but it would be good to have more.